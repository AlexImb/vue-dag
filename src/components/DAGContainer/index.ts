import DAGContainer from './DAGContainer.vue';

export { DAGContainer };
