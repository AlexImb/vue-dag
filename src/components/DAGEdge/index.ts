import DAGEdge from './DAGEdge.vue';

export { DAGEdge };
