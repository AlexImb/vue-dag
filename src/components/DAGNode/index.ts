import DAGNode from './DAGNode.vue';

export { DAGNode };
