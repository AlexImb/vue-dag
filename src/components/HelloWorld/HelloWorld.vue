<template>
  <div>
    <h1>Hello {{ msg }}</h1>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class HelloWorld extends Vue {
  @Prop() private msg!: string;
}
</script>
