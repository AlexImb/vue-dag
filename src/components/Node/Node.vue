<template>
  <div>
    <slot></slot>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class Node extends Vue {}
</script>
