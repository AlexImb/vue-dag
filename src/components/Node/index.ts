import Node from './Node.vue';

export { Node };
