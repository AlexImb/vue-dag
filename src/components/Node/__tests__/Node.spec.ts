import { shallowMount, Wrapper } from '@vue/test-utils';
import { Node } from '@/components/Node';
import { HelloWorld } from '@/components/HelloWorld';

let wrapper: Wrapper<Node>;

describe('DAGNode', () => {
  beforeEach(() => {
    wrapper = shallowMount(Node);
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders HelloWorld component inside the default slot', () => {
    wrapper = shallowMount(Node, {
      slots: { default: HelloWorld },
    });
    expect(wrapper.find(HelloWorld).isVueInstance()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
