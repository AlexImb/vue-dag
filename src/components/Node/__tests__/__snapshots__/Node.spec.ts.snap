// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DAGNode renders HelloWorld component inside the default slot 1`] = `
<div>
  <div>
    <h1>Hello </h1>
  </div>
</div>
`;

exports[`DAGNode renders correctly 1`] = `<div></div>`;
